desktopexporter/app/components/sidebar-view/trace-list.tsx

```diff
@@ -144,18 +145,50 @@ type TraceListProps = {
 export function TraceList(props: TraceListProps) {
   let ref = useRef(null);
   let size = useSize(ref);
+
   let location = useLocation();
+  let navigate = useNavigate();
+
+  let selectedIndex = 0;
+  let selectedTraceID = "";
   let { traceSummaries } = props;
 
   // Default to the first trace in the list if none are selected
-  let selectedTraceID = "";
   if (location.pathname.includes("/traces/")) {
     selectedTraceID = location.pathname.split("/")[2];
+    selectedIndex = traceSummaries.findIndex(
+      (summary) => summary.traceID === selectedTraceID,
+    );
   } else {
-    selectedTraceID = traceSummaries[0].traceID;
+    selectedTraceID = traceSummaries[selectedIndex].traceID;
     window.location.href = `/traces/${selectedTraceID}`;
   }
 
+  // Set up keyboard navigation
+  let arrowLeftPressed = useKeyPress("ArrowLeft");
+  let arrowRightPressed = useKeyPress("ArrowRight");
+  let hPressed = useKeyPress("h");
+  let lPressed = useKeyPress("l");
```

Comment:
As a hygiene thing, it would be good if the specific key didn't leak into the variable names. If we want to change them later, or support multiple layouts (not only other language layouts, but dvorak programmers a a thing) then we'll want this to be configurable.

So one more layer of indirection here would be my suggestion. Maybe a `keyMapping` type.

```typescript
type KeyMapping = {
  nextSpan: string;
  prevSpan: string;
  nextTrace: string;
  prevTrace: string;
  // support a second mapping, or maybe use a string[] as a value instead?
  nextSpanAlt: ?string;
  prevSpanAlt: ?string;
  nextTraceAlt: ?string;
  prevTraceAlt: ?string;
}
```

```typescript
let nextSpanPressed = useKeyPress(keyMapping.nextSpan);
```

You could also modify `useKeyPress` to accept an array of keys directly

```typescript
let prevTracePressed = useKeyPress(["ArrowLeft", "h"])
```

Reply:
Hey, so it was my understanding that using event.key in the `useKeyPress` hook (as opposed to event.keyCode) means this will work fine for Dvorak as is.

For the rest though, I agree with you. I'm going to modify `useKeyPress` to take arrays in a separate PR, so I can clean up navigation for trace summaries in the sidebar and the spans in the waterfall view at the same time. 😁